### PR TITLE
Remove 300ms delay before logoless boot

### PR DESCRIPTION
**Pull request by Fordi**
Tested on hardware, removing the delay clears up the boot noise problem in issue #9

### CODE REVIEW

**Comment by Fordi** [comment]
This PR eliminates all pre-boot screen noise for games with `disable_startup_logo`.

Findings:

Removing the delay mostly resolves it on hardware - though there's still a nigh-imperceptible one-frame flash of randomness..

I'm a little confused: Hoisting the `pen(0, 0, 0); clear();` out of the `#ifdef NO_STARTUP_LOGO` (so it hits both branches) does _not_ resolve the issue, even though it should, in theory.

Moving the `backlight(75);` into its own `#ifdef NO_STARTUP_LOGO` branch after the `sleep(300);`, also mostly resolves it, though with the same one-frame flash.

~~I was not able to get rid of the flash despite experimentation, and this is the simplest change that (mostly) succeeded, so I'm pulling this PR out of draft.~~

The missing piece was to flip the screen buffer in the `NO_STARTUP_LOGO` branch, then wait for the transfer to finish and vsync to settle out before flipping on the backlight.

**Comment by Fordi** [comment]
Pulling back to draft; I _think_ what's going on is that the screen buffers aren't getting initialized.

**Comment by Fordi** [comment]
Ok.  Clearing the screen and flipping twice did it, but only _after_ making it wait for the flips.

**Comment by Fordi** [comment]
I keep pulling back to draft, but it occurs to me there might be An Easier Way™.

**Comment by Fordi** [comment]
Final one.

Incidentally, the program used for testing is just:

```
#include "picosystem.hpp"
using namespace picosystem;

void init() {
    // Whatever's on screen stays on screen for a second
    sleep(1000);
    // Called once at the start, this is an ideal place to do any setup for your project.
}

void update(uint32_t tick) {
    // Called at a fixed rate - by default every 10ms. Perform any tasks that update the
    // game world state. Because it is called at a fixed rate you can easily time movement
    //  and events. tick is the number of update() calls since the game started.
}


void draw() {
    // Called as often as possible, aiming for 60 fps
}
```

File: libraries/picosystem.cpp [hunk 1]
```diff
@@ -20,11 +20,11 @@ namespace picosystem {
 
   #ifndef DYNAMIC_BUFFER
   #ifdef PIXEL_DOUBLE
-    color_t _fb[120 * 120];
+    color_t _fb[120 * 120] = { };
```
Comment (Fordi) [inline libraries/picosystem.cpp:23]: An empty initializer initializes to `0`'s.  No initializer leaves it randomized.

File: libraries/picosystem.cpp [hunk 2]
```diff
@@ -88,13 +88,19 @@ int main() {
...
   #else
+    // Keep the screen off...
+    backlight(0);
+    // Screen buffer is initialized clear; just flip it.
+    _wait_vsync();
+    _flip();
+    // Wait for the DMA transfer to finish
+    while (_is_flipping());
+    // wait for the screen to update
+    sleep(20);
```
Comment (Fordi) [inline libraries/picosystem.cpp]: Not sure why, but `_wait_vsync()` here gets The Flash™, while just forcing a delay doesn't.  \*shrug\*; there was a 300ms sleep before so whatever.

File: libraries/picosystem.cpp [hunk 3]
```diff
@@ -88,13 +88,19 @@ int main() {
...
   #else
+    // Keep the screen off...
+    backlight(0);
+    // Screen buffer is initialized clear; just flip it.
+    _flip();
+    // Wait for the DMA transfer to finish
+    while (_is_flipping());
+    // wait for the screen to update
+    _wait_vsync();
+    _wait_vsync();
```
Comment (Fordi) [inline libraries/picosystem.cpp:99]: Just one leaves The Flash™, two leaves it out.

**Comment by Gadgetoid** [comment]
Thank you!

File: libraries/picosystem.cpp (hunk 3 above)
Comment (Gadgetoid) [reply libraries/picosystem.cpp:99]: So it does! We'll probably dig deeper into this. Thank you!

**Pull request merged**